cosmos-abci/src/lib.rs

```diff
@@ -334,3 +445,111 @@ impl<T: Trait> sp_runtime::offchain::storage_lock::BlockNumberProvider for Modul
         <frame_system::Module<T>>::block_number()
     }
 }
+
+impl<T: Trait> Convert<T::AccountId, Option<T::AccountId>> for utils::StashOf<T> {
+    fn convert(controller: T::AccountId) -> Option<T::AccountId> {
+        let account_ledger: OptionalLedger<T::AccountId> =
+            <Module<T>>::account_ledgers(&controller);
+        match account_ledger {
+            Some(_ledger) => Some(_ledger.0),
+            None => Some(controller),
+        }
+    }
+}
+
+impl<T: Trait> Convert<T::AccountId, Option<utils::Exposure<T::AccountId, Balance>>>
+    for utils::ExposureOf<T>
+{
+    fn convert(_validator: T::AccountId) -> Option<utils::Exposure<T::AccountId, Balance>> {
+        Some(utils::Exposure {
+            total: 0,
+            own: 0,
+            others: vec![],
+        })
+    }
+}
+
+impl<T: Trait> pallet_session::SessionManager<T::AccountId> for Module<T> {
+    fn new_session(_new_index: SessionIndex) -> Option<Vec<T::AccountId>> {
+        let substrate_node_validators = <pallet_session::Module<T>>::validators();
+        debug::info!(
+            "Substrate validators after last on_finalize {:?}",
+            substrate_node_validators
+        );
+        // todo
+        // Get cosmos accounts & active validators from rocks_db storage.
+        let last_cosmos_validators: Vec<utils::CosmosAccountId> = vec![];
+        let mut new_substrate_validators: Vec<T::AccountId> = vec![];
+        for cosmos_validator_id in &last_cosmos_validators {
+            let substrate_account_id = <CosmosAccounts<T>>::get(&cosmos_validator_id);
+            if substrate_account_id.is_some() {
+                if let Some(full_substrate_account_id) = substrate_account_id {
+                    new_substrate_validators.push(full_substrate_account_id);
+                }
```

Comment:
Print WARNING message if we can`t find substrate account for the corresponding cosmos account